include/bluetooth/att.h

```diff
@@ -63,8 +63,53 @@ int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
 size_t bt_eatt_count(struct bt_conn *conn);
 
 #endif /* CONFIG_BT_TESTING */
+
+/** @brief Connect Enhanced ATT bearers
+ *
+ * Sends a series of Credit Based Connection Requests to connect @p num_channels.
+ * The peer may not have resources for @p num_channels bearers and a smaller amount
+ * may be created.
+ *
+ * @param conn The connection to send the request on
+ * @param num_channels The number of Enhanced ATT beaeres to request.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
+
 #endif /* CONFIG_BT_EATT */
 
+/** @brief ATT callback structure. */
+struct bt_att_cb {
+	/** @brief An ATT channel has been connected.
+	 *
+	 *  @param conn Connection object.
+	 *  @param cid The Channel ID of the connected ATT channel
+	 *  @param mtu The Maximum Transmission Unit of the connected ATT channel
+	 *  @param mps The Maximum PDU payload Size of the connected ATT channel
+	 */
+	void (*att_chan_connected)(struct bt_conn *conn, uint16_t cid, uint16_t mtu, uint16_t mps);
```

Comment:
Do we want to make this more generic so that it is easier to add information to the callback? For example pass a struct like this to the callback:
```c
struct bt_att_chan_info {
    uint16_t cid;
    uint16_t mtu;
    uint16_t mps;
    ...
}
```

Reply:
A structure sounds sensible. I don't know what the value the service/application has of knowing the CID, as I don't see what it can be used for.
The MTU is necessary, as the ATT_MTU needed may be application specific, or it is specified by a higher level application than the BCS (for example services).
I'm not sure it is possible to specify the ATT_MTU by a KConfig variable now, but it may be a case that the peer supports a lower MTU than the device requests.